src/dbus.c

```diff
@@ -342,14 +344,13 @@ static void on_get_server_information(GDBusConnection *connection,
 void signal_notification_closed(notification *n, enum reason reason)
 {
         if (reason < REASON_MIN || REASON_MAX < reason) {
-                fprintf(stderr, "ERROR: Closing notification with reason '%d' not supported. "
-                                "Closing it with reason '%d'.\n", reason, REASON_UNDEF);
+                LOG_W("Closing notification with reason '%d' not supported. "
+                      "Closing it with reason '%d'.", reason, REASON_UNDEF);
                 reason = REASON_UNDEF;
         }
 
         if (!dbus_conn) {
-                fprintf(stderr, "ERROR: Tried to close notification but dbus connection not set!\n");
-                return;
+                LOG_E("Unable to close notification: No DBus connection.");
```

Comment:
Where did the return statement go?

A bit out of scope but this usage of `LOG_E` doesn't immediately abort as the documentation suggests. I'm not sure about the correct way to go on this, either leave it to just return but `dbus_conn` being `NULL` is a pretty big issue so maybe just use `DIE()` instead?

Random thought: Since we're almost always going to abort when logging an error is there a reason to map `LOG_E` and not just `DIE()`?

Reply:
The `g_error` aborts the program with `SIGTRAP`, so it won't return anyways.

`DIE` exits with critical, but does not abort the program. It should be correctable by the user. The user may just have specified a wrong param.

But if the condition `dbus_conn == NULL` is true, I consider this a programming error, as the `on_name_lost` handler should have handled this already.